--- Makefile	2022-05-19 03:18:31.857703725 +0530
+++ Program_3_Kernel_Module.c	2022-05-19 03:22:53.882551492 +0530
@@ -1,7 +1,28 @@
-obj-m += Program_3_Kernel_Module.o
+/* 
 
-all:
-	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) modules
+Write a patch to panic the kernel. Capture panic message
+
+*/
+
+#include <linux/kernel.h>
+#include <linux/module.h>
+
+MODULE_LICENSE("GPL");
+MODULE_AUTHOR("Drupad Joshi");
+MODULE_DESCRIPTION("A Simple panic program");
+
+static int test_panic_init(void)
+{
+    printk(KERN_INFO"%s: In init\n", __func__);
+    panic("Hello Kernel I am causing the panic\n");
+    return 0;
+}
+
+static void test_panic_exit(void)
+{
+    printk(KERN_INFO"%s: In exit\n", __func__);
+}
+
+module_init(test_panic_init);
+module_exit(test_panic_exit);
 
-clean:
-	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) clean
